rush-exec/src/errors.rs

```diff
@@ -33,4 +99,7 @@ pub enum ExecutableError {
     FailedToParseStdout(String),
     #[error("Failed to parse executable stderr: {0}")]
     FailedToParseStderr(String),
+    /// This variant is a fallthrough, and you should generally prefer a more specific/human-readable error
+    #[error("{0}")]
+    OtherIoError(#[from] IoError),
```

Comment:
If this is not specific to executables or builtins, it may be worth thinking about either:
1. Making a separate error type called `FileSystemError`
2. Making an upstream error enum (`CommandError` or something) that has a `File` variant

Reply:
Another thing I forgot to delete :sweat_smile:

Right now it's basically just spitting out a string, but those concrete errors types are never used. I agree the work in this PR should probably be lifted somewhere higher, but I'm gonna leave that alone pending further discussion.